File: epi_judge_python/primitive_divide.py
def divide(x, y):
    # Implement this placeholder.
    result = 0
    power = 32
    while x >= y:
        while y << power > x:
            power -= 1
        result += 1 << power
        x -= y << power
    return result


from test_framework import test_utils_generic_main, test_utils

if __name__ == '__main__':
    test_utils_generic_main.generic_test_main("primitive_divide.tsv", divide)

File: epi_judge_cpp/primitive_divide.cc
int Divide(int x, int y) {
  // Implement this placeholder.
  int result = 0;
  short power = 32;
  while (x >= y) {
	while (1 << power > x) {
		power--;
	}
	x -= y << power;
	result += 1 << power;
  }
  return result;
}

#include "test_framework/test_utils_generic_main.h"

int main(int argc, char* argv[]) {
  std::vector<std::string> param_names{"x", "y"};
  generic_test_main(argc, argv, param_names, "primitive_divide.tsv", &Divide);
  return 0;
}
